Blog post announcing the CLI kickstart functionality (#4199)
FusionAuth/fusionauth-site: astro/src/content/blog/fusionauth-cli-kickstart.mdx

---
publish_date: 2026-04-02
title: Kickstart FusionAuth with the CLI
description: The FusionAuth CLI now comes with the ability to install and kickstart a FusionAuth instance directly from your command line.
authors: Bryan Robinson
image: /img/blogs/cli-kickstart/featured-image.png
categories: Product
tags: cli, automation, nodejs, emails, messages, themes, importing
excerpt_separator: "{/* more */}"
---
import Aside from '../../components/Aside.astro';

Since release, the [FusionAuth CLI](https://github.com/FusionAuth/fusionauth-node-cli) has helped users better manage FusionAuth instances. Common use cases include theme management, email template configuration, and lambda function editing.

But what if the CLI could also help you get started with FusionAuth? The latest release of the CLI does exactly that.

{/* more */}

## Install the CLI

```

> npm install @fusionauth/cli

Reviewer: ````suggestion
```console
npm install @fusionauth/cli
```
````

{/* more */}

## Install the CLI

```console
npm install @fusionauth/cli
```

To use the CLI, you'll need a working install of Docker. Follow the [Docker Desktop install guide](https://docs.docker.com/desktop/) to set that up.

## Kickstart your educational process

We want to make sure that every new user of FusionAuth can get up and running as quickly as possible. Testing FusionAuth should be more about integrating with your application than it is about configuring your authentication provider.

In order to get you there faster, we now have a new suite of commands to get you a simple, well-configured starting point for your projects.

By running the following command, you’ll get a FusionAuth Docker compose file and a Kickstart configuration file that contains all you need to be ready to implement authentication: a FusionAuth instance, an API key, and an application. All with one command instead of tedious configuration and setup.

The [Kickstart](https://fusionauth.io/docs/get-started/download-and-install/development/kickstart) process will initialize a FusionAuth instance with all the data provided by the CLI so that you don't need to set that yourself in the admin.

> ```

Reviewer: ````suggestion
```console
````

---
publish_date: 2026-04-02
title: Kickstart FusionAuth with the CLI
description: The FusionAuth CLI now comes with the ability to install and kickstart a FusionAuth instance directly from your command line.
authors: Bryan Robinson
image: /img/blogs/cli-kickstart/featured-image.png
categories: Product
tags: cli, automation, nodejs, emails, messages, themes, importing
excerpt_separator: "{/* more */}"
---
import Aside from '../../components/Aside.astro';

> The [FusionAuth CLI](https://github.com/FusionAuth/fusionauth-node-cli) is a great tool for doing deeper work within FusionAuth. Things like dealing with your themes, working with email templates, or dealing with our powerful lambda functions help you get further with FusionAuth.

Reviewer: ```suggestion
Since release, the [FusionAuth CLI](https://github.com/FusionAuth/fusionauth-node-cli) has helped users better manage FusionAuth instances. Common use cases include theme management, email template configuration, and lambda function editing.
```
Suggestion: remove some subjective/vague/repetitive language ('great', 'deeper', 'dealing with'; add 'since release' to make it more obvious that we're referring to the past in this introduction.

> What if the CLI could also help you get started with FusionAuth? The latest release of the CLI does just that.

Reviewer: ```suggestion
But what if the CLI could also help you get started with FusionAuth? The latest release of the CLI does exactly that.
```
Suggestion: add transition 'but' to smooth the step from the past...to the future!

{/* more */}

> ## Install the CLI (or upgrade)

Reviewer: ```suggestion
## Install the CLI
```

Simpler is better, I'd say.

```
npm install @fusionauth/cli

> ```

Reviewer: ````suggestion
```console
npm install @fusionauth/cli
````
suggestion: specify a language for nicer highlighting + uncopyable prompt `$` character (ditto for other commands in this blog post)

<Aside nodark="true" type="caution" title="Docker Required">
Using the CLI to kickstart a FusionAuth instance requires a local version of Docker in your command-line PATH.

> </Aside>

Reviewer: ```suggestion
To use the CLI, you'll need a working install of Docker. Follow the [Docker Desktop install guide](https://docs.docker.com/desktop/) to set that up.
```
Suggestion: remove this from an aside -- a lot of readers skip over asides, and they break up the reading flow, anyway. I'd also suggest lifting this up above the code snippet, so users don't copy/paste/run that command, then hit an error.

Up to you if we link to Docker Desktop or some other install documentation -- I think we're probably targeting mostly macOS/Windows/Linux developers here, so that seems appropriate?

## Kickstart your educational process

> We want to make sure that every new user of FusionAuth can get up and running as quickly as possible. Testing FusionAuth should be more about integrating with your application than it is about configuring your authentication provider.

Reviewer: Suggestion: lift the code snippet + code snippet intro to the beginning of this section. Otherwise I'm not sure readers will understand what this section is about when they first read through -- you're burying the lede!

In order to get you there faster, we now have a new suite of commands to get you a simple, well-configured starting point for your projects.

By running the following command, you’ll get a FusionAuth Docker compose file and a Kickstart configuration file that contains all you need to be ready to implement authentication: a FusionAuth instance, an API key, and an application. All with one command instead of tedious configuration and setup.

The [Kickstart](https://fusionauth.io/docs/get-started/download-and-install/development/kickstart) process will initialize a FusionAuth instance with all the data provided by the CLI so that you don't need to set that yourself in the admin.

```
npx fusionauth kickstart:install
```

![Output of the kickstart:install command showing the setting of username, password, and application name, and how to access the newly installed FusionAuth](/img/blogs/cli-kickstart/cli-output.png)

> Once your FusionAuth configuration is ready, you can move into the created directory and run commands to deal with the Docker container.

Reviewer: ```suggestion
Use these commands from the new directory to manage your new FusionAuth Docker setup.
```
suggestion: simplify this a bit!

* `npx fusionauth kickstart:start`: starts your FusionAuth container and configures your instance
* `npx fusionauth kickstart:stop`: non-destructively pauses your FusionAuth container 
* `npx fusionauth kickstart:kill`: shuts down and removes your FusionAuth container

## The future of the CLI

> Developers should be in whatever environment works best for them. For the most part, that’s head-down writing code in their IDE. What better way to keep you focused and productive than to keep you from having to load up the FusionAuth dashboard just to configure your application.

Reviewer: ```suggestion
Developers should use whatever environment works best for them. For the most part, that’s writing code. What better way to stay focused than using the command line as much as possible?
```
Suggestion: the last sentence is a bit long and complicated, maybe we can simplify?

> As we move forward with new features for the CLI, be sure to let us know what would keep you moving forward with your applications by putting any requests in the issues for [the CLI on GitHub](https://github.com/FusionAuth/fusionauth-node-cli).

Reviewer: ```suggestion
As we improve the CLI, let us know what you'd like to see! You can file requests in the [CLI repo's GitHub Issues](https://github.com/FusionAuth/fusionauth-node-cli).
```
Suggestion: use fewer word when do trick
